content/using/running/vere.md

```diff
@@ -0,0 +1,632 @@
+
+- Undocked: `urbit grab [pier]`
+- Docked: `[pier]/.run grab`
+
+### `info`
+
+Print pier information.
+
+- Undocked: `urbit info [pier]`
+- Docked: `[pier]/.run info`
+
+```
+> urbit info zod
+loom: mapped 2048MB
+boot: protected loom
+live: loaded: MB/268.173.312
+boot: installed 351 jets
+
+urbit: zod at event 256133
+  disk: live=&, event=256133
+
+lmdb info:
+  map size: 1099511627776
+  page size: 4096
+  max pages: 268435456
+  number of pages used: 1506786
+  last transaction ID: 255733
+  max readers: 126
+  number of readers used: 0
+  file size (page): 6171795456
+  file size (stat): 6171795456
+```
+
+### `meld`
+
+Deduplicate ship state. This can significantly reduce memory usage for ships
+with large states. This is a common solution when ships use up all available
+mapped memory (currently 2GB) and crash with `bail: meme`.
+
+Note this command may use a large amount of memory during execution, up to
+around 8GB or so, depending on the size of the ship's state and other factors.
+If you do not have enough memory to run it, you may need to allocate some swap.
+
+If the meld succeeds, it will print out a memory usage report. If it exits
+saying `KILLED`, it means it ran out of memory and the operation was aborted.
+
+- Undocked: `urbit meld [pier]`
+- Docked: `[pier]/.run meld`
+
+### `pack`
+
+Defragment a ship's state. This will reduce the size of a ship's state much less
+than `meld`, but it is much faster and requires much less memory to complete.
+
+- Undocked: `urbit pack [pier]`
+- Docked: `[pier]/.run pack`
+
+### `prep`
+
+Prepare a pier for upgrade. This utility is designed to be a general-purpose
+forward-compatibility mechanism. What action (if any) is taken depends on the
+old version and new version. Currently, it just makes sure the snapshot is fully
+up-to-date.
+
+- Undocked: `urbit prep [pier]`
+- Docked: `[pier]/.run prep`
+
+### `next`
+
+Request a runtime upgrade. If your binary is already the latest version, no
+action will be taken. If a new binary is available, it will be upgraded.
+
+- Undocked: `urbit next [pier]`
+- Docked: `[pier]/.run next`
+
+### `queu AT-EVENT`
+
+Load a snapshot jamfile. `AT-EVENT` is an event number. There must be
+corresponding snapshot in `[pier]/.urb/roc/[AT-EVENT].jam` created
+previously with `cram` or `-n`.
+
+- Undocked: `urbit queu [pier] 10000`
+- Docked: `[pier]/.run queu 10000`
+
+### `vere ARGS DIR`
+
+Download a binary. `DIR` is an output directory (it must already exist) and `ARGS` are:
+
+- `-a, --arch ARCH` - architecture, `ARCH` may be one of `x86_64-linux`,
+  `x86_64-darwin`, and `x86_64-windows`, though more may be added in the future.
+- `-v, --version VER` - version number, `VER` is e.g. `1.9`.
+- `-p, --pace` - release channel, e.g. `live`.
+
+Example usage:
+
+- Undocked: `urbit vere -a x86_64-linux -v 1.9 -p live .`
+- Docked: `[pier]/.run vere -a x86_64-linux -v 1.9 -p live .`
+
+### `serf ARGS`
```

Comment:
At some point in this doc want an aside about what king v. serf are responsible for.

Reply:
I have added an explanation into the second paragraph at the top (king = I/O & event persistence, serf = computation & state persistence)